### DISCUSSION TITLE
Create new issues directly from projects list, not "draft" issues

### DISCUSSION THREAD

**Discussion by dlegr250** [community/community#30471] [Projects and Issues] [answered]
When I'm viewing my project's issue board/table, I almost always want to create a new issue from this view, not a "draft" issue. What I have to do right now, is:

1. Type out the title of the new issue
2. Select the drop-down arrow on far-left of new row > select "Convert to Issue" 
3. Type in the name of the project I want to add this issue to
4. Then add assignee/labels/etc now that the record exists

What I would LIKE to do is:
1. Type out title, assignee, labels, project, etc, in the field and hit enter, and new issue is created (not as a "draft" issue, but as an actual issue)

Linear.app is a project/issue management tool that does this very well. Suggesting taking a look at them for some good ideas.

I have created several dozen issues over the last few months for my projects, and every single one of them went through this process; not a single one of them was a "draft" issue. I'm sure there's a use for the draft issue, but I haven't found it for my team, but I can't imagine that most teams are creating a large number of draft issues that go through some process before they assign and label the task on a project?

Also, it would be nice to have a default project or assignee option as a setting, because 99% of the time I'm in the context of a single project/repo, and it is frustrating to have to assign every new issue to the project I'm currently working in.

Overall I'm happy with the GH issues and improvements over the last few months! I'm just trying to make it better so I can continue to enjoy it :)

**Comment by Marv93** [comment]
I also don't understand the use of draft issues and would like to create "real" issues directly. I could imagine that a viable solution would be to create some issue templates and when creating an issue in a project, one of those template issues can be selected. So the workflow would be to type in the title, select the template (via the plus symbol to the left of the title input field for example) and hitting enter. And in the template you could set the default assignee(s), label(s), ...

**Comment by cristeahub** [comment]
I agree with this. Not having the option to create an issue directly is the last major blocker for us to move to github projects fulltime!

**Reply by PhABC** [reply]
Agreed, same

**Reply by jeffgrimes9** [reply]
Agreed, the extra clicks are annoying given that we are often filing dozens of new issues in a day

**Reply by thomaze1988** [reply]
Same

**Comment by michaelmccallam** [comment]
@dlegr250 The drafts 'feature' can be a hand concept for the following scenarios. 


1. Drafts are visible on the project and not the repo
2. Issues on a repo can't be deleted.

1. Once you create an issue on a repo, the object model is tied to the repo rather than the project. So you can't remove issues, only close them. It's useful to remove them completely IF they contain any internal only information or are test issues or simply offer no value. By adding a draft you can still use them on the full SDLC through the board to Done, and keep them visible only on the project and not on the repo. 
This is especially useful for public repos where developers are both internal company contributors and external FOSS contributors.

**Reply by JLScribani** [reply]
This would be an acceptable solution only if you were also able to add comments, labels, etc to drafts so that they can be context-rich for project management purposes. A draft with a title is useless.

**Comment by rileybroughten** [comment, marked as answer]
👋 We recently shipped the ability to create issues directly from a project, check out the announcement [here](https://github.blog/changelog/2023-01-26-github-projects-january-26th-update/#create-issues-in-a-snap-with-the-new-issue-creation-dialog) as well as the [documentation](https://docs.github.com/en/issues/planning-and-tracking-with-projects/managing-items-in-your-project/adding-items-to-your-project#creating-issues) and let us know what you think!

**Comment by dlegr250** [comment]
@rileybroughten That's awesome! I just tested it out, it does exactly what I wanted. I hope this doesn't sound nit-picky, but I do have a few feature requests to build upon this. I use Github issues a lot, so want a few minor improvements:

1. Hotkey support? To create a new issue I have to use my mouse to click the "+" button, then select "Create new issue". Would love a hotkey or a dedicated "New issue" button on the screen. It looks like some of the fields have hotkeys listed on them (a for assignee), but those don't seem to work for me?
2. Issue value defaults? On each issue, I often assign myself. What about a setting or something that allows you to default new tickets to assign to yourself?

Definitely a great improvement, looking forward to new features in Github issues!

**Reply by rileybroughten** [reply]
Thanks for this feedback! Tagging @j0siepy on here as well for eyes on the feedback and will make sure this gets passed along to the team! 👍

**Comment by PhABC** [comment]
Would there be any way to make this the default behaviour, perhaps in project board settings? Such that we can just type and press Enter like it's done for drafts?

**Reply by rileybroughten** [reply]
@PhABC if you type `#` to search for and select a repository, you can create a new issue while typing directly from the omnibar! That allows you to title the issue and supply some starting information, rather than needing to create a draft first and then convert it.

https://user-images.githubusercontent.com/101840513/224179420-c489ef9d-ed33-4f5c-94b7-9e4b4e8d6080.mov

**Reply by PhABC** [reply]
Very nice! That's perfect

**Reply by danielracca** [reply]
It would be nice if we could select a preferred default repository per project and then click “+” would automatically always add an issue.

**Reply by jthantranon** [reply]
^ this. also using the # after hitting enter you still have to select a "template" which is more buttons to click. Same amount as making a draft then converting it.

**Reply by cyberw** [reply]
@rileybroughten Any updates? Would be really nice to have :)

**Comment by eliorpaz** [comment]
Hi 

Is there an option to create new issues from project using API?

**Comment by FunkyMuse** [comment]
Is there an ability to create an item and have it as a non-draft, for example we have designers on our team and we're using Github projects, but their project isn't linked to any issue on a repository...

**Comment by romanr** [comment]
Nobody asked for this Draft issues. I create using "+" in omnibar as suggested and Draft is created. Have to change to issue every time and on top of everything add to project every time, while  Iam already in the project creating the issue. What is this, project management for bureaucrats?
